moveit_ros/moveit_servo/include/moveit_servo/servo_calcs.h

```diff
@@ -296,7 +300,8 @@ class ServoCalcs
   sensor_msgs::msg::JointState internal_joint_state_, original_joint_state_;
   std::map<std::string, std::size_t> joint_state_name_map_;
 
-  std::vector<LowPassFilter> position_filters_;
+  // Smoothing algorithm (loads a plugin)
+  std::shared_ptr<online_signal_smoothing::SmoothingBaseClass> smoother_;
```

Comment:
unique_ptr

Reply:
I think it has to be a shared_ptr to work as a plugin. unique_ptr causes this:

> error: no match for ‘operator=’ (operand types are ‘std::unique_ptr<online_signal_smoothing::SmoothingBaseClass>’ and ‘std::shared_ptr<online_signal_smoothing::SmoothingBaseClass>’)
>   179 |     smoother_ = smoothing_loader_.createSharedInstance(parameters_->smoothing_filter_plugin_name);

Reply:
There is `createUniqueInstance()` or `createUnmanagedInstance()` for this.

Reply:
No luck.

createUniqueInstance:

```
error: no match for ‘operator=’ (operand types are ‘std::unique_ptr<online_signal_smoothing::SmoothingBaseClass>’ and ‘pluginlib::UniquePtr<online_signal_smoothing::SmoothingBaseClass>’ {aka ‘std::unique_ptr<online_signal_smoothing::SmoothingBaseClass, std::function<void(online_signal_smoothing::SmoothingBaseClass*)> >’})
  179 |     smoother_ = smoothing_loader_.createUniqueInstance(parameters_->smoothing_filter_plugin_name);
```

createUnmanagedInstance:

```
error: no match for ‘operator=’ (operand types are ‘std::unique_ptr<online_signal_smoothing::SmoothingBaseClass>’ and ‘online_signal_smoothing::SmoothingBaseClass*’)
  179 |     smoother_ = smoothing_loader_.createUnmanagedInstance(parameters_->smoothing_filter_plugin_name);
```

Reply:
well, you would have to either store the UniquePtr type (createUniqueInstance) or reset the unique_ptr using the raw pointer returned by `createUnmanagedInstance`. But anyway, it's not a big deal.